astropy/visualization/lupton_rgb.py

```diff
@@ -122,238 +227,108 @@ def intensity(self, image_r, image_g, image_b):
-        self._slope = frac * self._uint8Max / np.arcsinh(frac * Q)
-
-        self._soften = Q / float(stretch)
-
-    def map_intensity_to_uint8(self, I):
-        # n.b. np.where can't and doesn't short-circuit
-        with np.errstate(invalid="ignore", divide="ignore"):
-            return np.where(I <= 0, 0, np.arcsinh(I * self._soften) * self._slope / I)
-
-
-class AsinhZScaleMapping(AsinhMapping):
-    """
-    A mapping for an asinh stretch, estimating the linear stretch by zscale.
-
-    x = asinh(Q (I - z1)/(z2 - z1))/Q
-
-    Parameters
-    ----------
-    image1 : ndarray or a list of arrays
-        The image to analyse, or a list of 3 images to be converted to
-        an intensity image.
-    image2 : ndarray, optional
-        the second image to analyse (must be specified with image3).
-    image3 : ndarray, optional
-        the third image to analyse (must be specified with image2).
-    Q : float, optional
-        The asinh softening parameter. Default is 8.
-    pedestal : float or sequence(3), optional
-        The value, or array of 3 values, to subtract from the images; or None.
-
-    Notes
-    -----
-    pedestal, if not None, is removed from the images when calculating the
-    zscale stretch, and added back into Mapping.minimum[]
-    """
-
-    def __init__(self, image1, image2=None, image3=None, Q=8, pedestal=None):
-        if image2 is None or image3 is None:
-            if not (image2 is None and image3 is None):
-                raise ValueError(
-                    "please specify either a single image or three images."
-                )
-            image = [image1]
-        else:
-            image = [image1, image2, image3]
-
-        if pedestal is not None:
-            try:
-                len(pedestal)
-            except TypeError:
-                pedestal = 3 * [pedestal]
-
-            if len(pedestal) != 3:
-                raise ValueError("please provide 1 or 3 pedestals.")
-
-            image = list(image)  # needs to be mutable
-            for i, im in enumerate(image):
-                if pedestal[i] != 0.0:
-                    image[i] = im - pedestal[i]  # n.b. a copy
-        else:
-            pedestal = len(image) * [0.0]
-
-        image = compute_intensity(*image)
-
-        zscale_limits = ZScaleInterval().get_limits(image)
-        zscale = LinearMapping(*zscale_limits, image=image)
-        # zscale.minimum is always a triple
-        stretch = zscale.maximum - zscale.minimum[0]
-        minimum = zscale.minimum
-
-        for i, level in enumerate(pedestal):
-            minimum[i] += level
-
-        AsinhMapping.__init__(self, minimum, stretch, Q)
-        self._image = image
+        return np.asarray(image_rgb)
 
 
 def make_lupton_rgb(
-    image_r, image_g, image_b, minimum=0, stretch=5, Q=8, filename=None
+    image_r,
+    image_g,
+    image_b,
+    minimum=0,
+    stretch=5,
+    Q=8,
+    filename=None,
+    stretch_object=None,
```

Comment:
Remove, since we don't need to actually customize the stretch class, just the interval.

Reply:
As noted in the other comments, as was originally written in the make_lupton_rgb.py code, the ``LuptonAsinhZscaleStretch`` stretch uses a Zscale calculation internally. This isn't solved by just adding an ``interval`` input to ``make_lupton_rgb()`` (which instead removes the need for a ``minimum`` keyword argument). 

The entire ``LuptonAsinhZscaleStretch`` class could be removed, and the user would then have to self-calculate the linear ``stretch`` variable estimation to replicate the Zscale behavior for use directly with ``LuptonAsinhStretch``.

However, if ``make_lupton_rgb()`` takes a stretch class instance, this allows for arbitrary stretch schemes to be used with the Lupton et al. (2004) intensity calculation (which is an independent step from the choice of normalization from the interval class, and the scaling from the stretch class).